lib/src/core/configs/chip_config.dart

```diff
@@ -5,30 +5,27 @@ import 'package:json_theme/json_theme.dart';
 class ChipConfig extends ComponentConfig {
   const ChipConfig({
     required this.theme,
-    required this.iconTheme,
-    required this.surfaceTintColor,
-    required this.deleteIconColor,
     required this.clipBehaviour,
     required this.visualDensity,
     required this.materialTapTargetSize,
     required this.autofocus,
   });
 
   final ChipThemeData? theme;
-  final IconThemeData? iconTheme;
-  final Color? surfaceTintColor;
-  final Color? deleteIconColor;
+
   final Clip clipBehaviour;
   final VisualDensity? visualDensity;
   final MaterialTapTargetSize? materialTapTargetSize;
   final bool autofocus;
 
   factory ChipConfig.fromMap(Map<String, dynamic> map) {
     return ChipConfig(
-      theme: ThemeDecoder.decodeChipThemeData(map['theme']),
-      iconTheme: ThemeDecoder.decodeIconThemeData(map['iconTheme']),
-      surfaceTintColor: ThemeDecoder.decodeColor(map['surfaceTintColor']),
-      deleteIconColor: ThemeDecoder.decodeColor(map['deleteIconColor']),
+      theme: (map['theme'] != null)
+          ? (map['theme']['backgroundColor'] != null &&
+                  map['theme']['brightness'] != null)
+              ? ThemeDecoder.decodeChipThemeData(map['theme'])
+              : const ChipThemeData()
+          : const ChipThemeData(),
```

Comment:
What if theme has other parameters value present except background color and brightness? This will override the present value as well. Also no need to provide a default value when the whole theme object is null.